controllers/sriovoperatorconfig_controller.go

```diff
@@ -229,11 +230,7 @@ func (r *SriovOperatorConfigReconciler) syncWebhookObjs(ctx context.Context, dc
 		data.Data["CaBundle"] = os.Getenv("WEBHOOK_CA_BUNDLE")
 		data.Data["DevMode"] = os.Getenv("DEV_MODE")
 		data.Data["ImagePullSecrets"] = GetImagePullSecrets()
-		external, err := utils.IsExternalControlPlaneCluster(r.Client)
-		if err != nil {
-			logger.Error(err, "Fail to get control plane topology")
-			return err
-		}
+		external := r.OpenshiftContext.IsHypershift()
```

Comment:
I suspect this can cause panic:

https://go.dev/play/p/hDtR5cPY6Lj?v=goprev

Maybe it's better to change `IsHypershift()`'s receiver from `OpenshiftContext` to `*OpenshiftContext` and add a nil check inside that function.

Reply:
I'm not in favor of passing around pointers just to check for nil. Instead, I think we should pass around the context still, and have it explicitly say it is not Openshift.

Reply:
good catch!
I update to first check if we are running on ocp